Taarafo.Core.Tests.Unit/Services/Foundations/PostImpressions/PostImpressionServiceTests.Validations.Add.cs

```diff
@@ -0,0 +1,206 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Moq;
+using Taarafo.Core.Models.PostImpressions;
+using Taarafo.Core.Models.PostImpressions.Exceptions;
+using Xunit;
+
+namespace Taarafo.Core.Tests.Unit.Services.Foundations.PostImpressions
+{
+    public partial class PostImpressionServiceTests
+    {
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnAddIfPostImpressionIsNullAndLogItAsync()
+        {
+            //given
+            PostImpression nullPostImpression = null;
+
+            var nullPostImpressionException =
+                new NullPostImpressionException();
+
+            var expectedPostImpressionValidationException =
+                new PostImpressionValidationException(nullPostImpressionException);
+
+            //when
+            ValueTask<PostImpression> addPostImpressionTask =
+                this.postImpressionService.AddPostImpressions(nullPostImpression);
+
+            //then
+            await Assert.ThrowsAsync<PostImpressionValidationException>(() =>
+                addPostImpressionTask.AsTask());
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedPostImpressionValidationException))),
+                        Times.Once);
+
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnAddIfPostImpressionIsInvalidAndLogItAsync()
+        {
+            //given
+            Guid invalidGuid = Guid.Empty;
+
+            var invalidPostImpression = new PostImpression
+            {
+                PostId = invalidGuid,
+                ProfileId = invalidGuid
+            };
+
+            var invalidPostImpressionException =
+                new InvalidPostImpressionException();
+
+            invalidPostImpressionException.AddData(
+                key: nameof(PostImpression.PostId),
+                values: "Id is required.");
+
+            invalidPostImpressionException.AddData(
+                key: nameof(PostImpression.Post),
+                values: "Post is required.");
```

Comment:
Message should be as `Object is required`

Reply:
Changed the message in the commit named CODERUB: Changed Message For Profile And Post

Reply:
The Pull Request with the changes can be found here: https://github.com/hassanhabib/Taarafo.Core/pull/221